orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy

```diff
@@ -112,19 +112,25 @@ class BakeStage implements BranchingStageDefinitionBuilder, RestartableStage {
       def bakeInitializationStages = stage.execution.stages.findAll {
         it.parentStageId == stage.parentStageId && it.status == ExecutionStatus.RUNNING
       }
-
+      def imageProducingStages = stage.execution.stages.findAll {
+        it.type == PIPELINE_CONFIG_TYPE && bakeInitializationStages*.id.contains(it.parentStageId) && (it.context.ami || it.context.imageId)
+      }
       def globalContext = [
-        deploymentDetails: stage.execution.stages.findAll {
-          it.type == PIPELINE_CONFIG_TYPE && bakeInitializationStages*.id.contains(it.parentStageId) && (it.context.ami || it.context.imageId)
-        }.collect { Stage bakeStage ->
+        deploymentDetails: imageProducingStages.collect { Stage bakeStage ->
           def deploymentDetails = [:]
           ["ami", "imageId", "amiSuffix", "baseLabel", "baseOs", "refId", "storeType", "vmType", "region", "package", "cloudProviderType", "cloudProvider"].each {
             if (bakeStage.context.containsKey(it)) {
               deploymentDetails.put(it, bakeStage.context.get(it))
             }
           }
-
           return deploymentDetails
+        },
+        artifacts: imageProducingStages.collect { Stage bakeStage ->
+          def artifact = [:]
+          if (bakeStage.context.containsKey("artifact")) {
```

Comment:
Yup, or if you tag multiple docker images in your build (one per release track or various global registries).